File: soal10.py
def twoSum(nums, target):
    num_map = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], i]
        num_map[num] = i

# Contoh penggunaan
if __name__ == "__main__":
    nums = list(map(int, input("Masukkan angka dalam array: ").split()))
    target = int(input("Masukkan target: "))
    print(twoSum(nums, target))

File: soal10.ts
function twoSum(nums: number[], target: number): number[] {
    let numMap: Map<number, number> = new Map();

    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i];
        if (numMap.has(complement)) {
            return [numMap.get(complement)!, i];
        }
        numMap.set(nums[i], i);
    }

    return []; // Harusnya tidak terjadi karena diasumsikan ada satu solusi
}

// Contoh penggunaan
const inputNums: string | null = prompt("Masukkan angka dalam array (dipisahkan dengan spasi): ");
const target: number = parseInt(prompt("Masukkan target: ") || "0");

if (inputNums) {
    const nums: number[] = inputNums.split(" ").map(Number);
    console.log(twoSum(nums, target));
}
